src/NHibernate/Loader/Criteria/CriteriaQueryTranslator.cs

```diff
@@ -863,6 +863,24 @@ public string GetEntityName(ICriteria subcriteria, string propertyName)
 			return GetEntityName(subcriteria);
 		}
 
+		internal IDictionary<string, string> GetCriteriaSQLAliasMap()
+		{
+			var result = outerQueryTranslator != null
+				? new Dictionary<string, string>(outerQueryTranslator.GetCriteriaSQLAliasMap())
+				: new Dictionary<string, string>();
```

Comment:
We can avoid creating a new dictionary with in 2 ways:

1. Replace it with a `.Concat`. The downside is that the combined collection would contain duplicate elements. So it will depend on the order in which the aliases are rendered.
2. [Encapsulate](https://github.com/hazzik/nhibernate-core/pull/109) logic of rendering of SQL aliases here. This solves both creating a new dictionary and order of rendering aliases as it would be explicitly coded.